File: java/merge_sort/src/test/java/com/ahmedsameha1/edx_algorithms_and_data_structures/merge_sort/TestMergeSort.java
package com.ahmedsameha1.edx_algorithms_and_data_structures.merge_sort;

// Copied from: https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html
import static org.junit.jupiter.api.Assertions.*;
// Copied from: https://junit.org/junit5/docs/current/user-guide/
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

public class TestMergeSort {
    private MergeSort mergeSort;

    @BeforeEach
    public void setup() {
        mergeSort = new MergeSort();
    }

    @Test
    public void test() {
        assertArrayEquals(new int[]{1, 2, 3, 4, 5}, mergeSort.sort(5, 4, 3, 2, 1));
        assertArrayEquals(new int[] {1, 2, 3, 4}, mergeSort.sort(4, 3, 2, 1));
        assertArrayEquals(new int[] {-1, 1, 2, 3, 4}, mergeSort.sort(4, 3, 2, 1, -1));
        assertArrayEquals(new int[] {-1, 15, 15, 52, 73, 104}, mergeSort.sort(15, 73, 104, 15, 52, -1));
    }
}

File: python/merge_sort/test_merge_sort.py
import unittest
import merge_sort

class TestMergeSort(unittest.TestCase):
    def test_merge_sort(self):
        self.assertListEqual(merge_sort.sort([23, 14, 62, -1, 79, 40, 237, 136, -78]),[-78, -1, 14, 23, 40, 62, 79, 136, 237])
        self.assertListEqual(merge_sort.sort([4, 3, 2, 1]), [1, 2, 3, 4])

if __name__ == "__main__":
    unittest.main()
